crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -90,8 +90,11 @@ impl ModuleSubscriptions {
     }
 
     pub fn remove_subscriber(&self, client_id: ClientActorId) {
-        let mut subscriptions = self.subscriptions.write();
-        self._remove_subscriber(client_id, &mut subscriptions);
+        let this = self.clone();
+        tokio::task::spawn_blocking(move || {
+            let mut subscriptions = this.subscriptions.write();
+            this._remove_subscriber(client_id, &mut subscriptions);
+        });
```

Comment:
I think it'd be better to do this spawn_blocking in remove_subscriber in module_host, not here. And make sure you `.await` on the spawned task, otherwise this becomes asynchronous which I don't think is what we want(?)

Reply:
Makes sense!